src/TorchSharp/DisposeScopeManager.cs

```diff
@@ -42,15 +42,15 @@ internal static DisposeScope NewDisposeScope()
 
         internal void RemoveDisposeScope(DisposeScope disposeScope)
         {
-            Debug.Assert(DisposeScopeStack.Count > 0);
-            Debug.Assert(DisposeScopeStack.Peek() == disposeScope);
-            DisposeScopeStack.Pop();
+            var index = DisposeScopeStack.LastIndexOf(disposeScope);
+            if (index is not -1)
```

Comment:
I think it should be an error (exception) if you are removing a dispose scope that isn't registered. The previous code treated it as an assertion, but I think the right behavior is to throw.

Reply:
This method is called by dispose scope's `Dipose`. If the dispose scope is not in that stack, it means that it has been disposed previously, so doing nothing is what we expect.

Reply:
And could you please check my comment [here](https://github.com/dotnet/TorchSharp/pull/1304#issuecomment-2088234426). This is also about the "relationship" between `DisposeScope` and `DisposeScopeManager`.

Reply:
Yes, I note that it's an `internal` method, so it should be okay.

Anyway, I'm going to be busy with some private matters today. I'll merge the two PRs that were accepted later today.

Reply:
BTW, since you have made significant contributions to the repo, it would be great if you would introduce yourself a little bit in this thread:

https://github.com/dotnet/TorchSharp/discussions/333

It's always nice to know a little bit about the people who build the software, and the people you work with on it.

The same goes for you, @shaltielshmid... :-)